pkg/apis/infrastructure/v1alpha2/azuremachine_types.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// AzureMachineSpec defines the desired state of AzureMachine
+type AzureMachineSpec struct {
+	Location         string `json:"location"`
+	AvailabilityZone string `json:"availabilityZone"`
+	VMSize           string `json:"vmSize"`
+	Image            Image  `json:"image"`
+	OSDisk           OSDisk `json:"osDisk"`
+	SSHPublicKey     string `json:"sshPublicKey"`
```

Comment:
AzureMachineSpec will need a ProviderID field that is used by the CAPI Machine controller for linking back to the related Node resource that the Machine creates.